src/test/java/com/checkmarx/flow/cucumber/integration/publishprocess/RunPublishProcessSteps.java

```diff
@@ -0,0 +1,428 @@
+
+    private FindingsType findingsType;
+
+    private List<Filter> filters;
+
+    private boolean needFilter;
+
+    private int totalResults;
+
+    private boolean useSanityFindingsFile = false;
+
+    private boolean useOneFindingForUpdateClose = false;
+
+    private long updateTime;
+
+    private String issueUpdateVulnerabilityType;
+    private String issueUpdateFilename;
+    private Long issueId;
+
+    private boolean updating;
+
+
+    @Before("@PublishProcessing")
+    public void setDefaults() {
+        needFilter = false;
+        useSanityFindingsFile = false;
+        updating = false;
+        issueUpdateVulnerabilityType = "";
+        issueUpdateFilename = "";
+        useOneFindingForUpdateClose = false;
+    }
+
+    @Before("@PublishProcessing")
+    public void setOfflineMode() {
+        cxProperties.setOffline(true);
+    }
+
+    @Before("@PublishProcessing")
+    public void cleanJiraProject() throws IOException {
+        jiraUtils.ensureProjectExists(jiraProperties.getProject());
+        //jiraUtils.ensureIssueTypeExists(jiraProperties.getIssueType());
+        jiraUtils.cleanProject(jiraProperties.getProject());
+    }
+
+    @Given("target is JIRA")
+    public void setTargetTypeToJira() {
+        bugTracker = BugTracker.Type.JIRA;
+    }
+
+    @Given("there are {int} results from which {int} results match the filter")
+    public void setResultsAndFilters(int totalResults, int matchingResults) {
+        findingsType = FindingsType.DIFFERENT_SEVERITIES;
+        numOfFindings = matchingResults;
+        this.totalResults = totalResults;
+        needFilter = true;
+        Filter filter = Filter.builder().type(Filter.Type.SEVERITY).value("High").build();
+        filters =Arrays.asList(filter);
+    }
+
+    @Given("filter-severity is {}")
+    public void setSeverityFilterTypes(String types) {
+        filters = createFiltersFromString(types, Filter.Type.SEVERITY);
+        needFilter = true;
+    }
+
+    @Given("using sanity findings")
+    public void setUseSanityFindingsFile() {
+        useSanityFindingsFile = true;
+    }
+
+    @Given("there is an existing issue")
+    public void createExistingIssueForUpdate() throws IOException, ExitThrowable, InterruptedException {
+        ScanRequest request = getScanRequestWithDefaults();
+        File file = getFileFromResourcePath("cucumber/data/sample-sast-results/1-finding-create-for-update.xml");
+        innerPublishRequest(request, file);
+        updateTime = jiraUtils.getIssueUpdatedTime(jiraProperties.getProject());
+        issueId = jiraUtils.getFirstIssueId(jiraProperties.getProject());
+        issueUpdateVulnerabilityType = jiraUtils.getIssueVulnerability(jiraProperties.getProject());
+        issueUpdateFilename = jiraUtils.getIssueFilename(jiraProperties.getProject());
+        Assert.assertTrue("Issue priority before update is incorrect",assertIssuePriority(ISSUE_PRIORITY_BEFORE_UPDATE));
+        TimeUnit.SECONDS.sleep(2);
```

Comment:
This may make the test fragile.

Reply:
If it is about the sleep, it was reviewed by Aviv (If I am not mistaken), and it's ok, otherwise the update is too rapid, and there is no time difference between the creation time, and the update time. 2 seconds sleep will always create a difference.